docs/proposal-multi-user-approach.md

```diff
@@ -13,11 +12,344 @@ via a UI Server running as the suite owner for UI purposes meaning that, from
+
+E.g.  Suppose User3 is a member of Group3:
+
+```python
+c.UIServer.authorisation = {
+    "User3": ["READ","!ADMIN", "poll"],               # Specified interactions granted to user3
+    "group:Group3": ["READ","ADMIN"],                 # Both READ and ADMIN access to workflows
+                                                      # granted to users in group1
+}
+```
+
+User3 has operations `READ`,`ADMIN` added through membership of Group3 and then `ADMIN` operations are removed by specific User3 config, resulting in User3 having only `READ` access. The addition of `poll` will have no effect. This is due to negations taking preference, and will need to be documented.
+
+## Possible Access Assignment of Mutations and Queries
+
+### Queries
+
+Proposal: All queries can be executed if the user has `Read` access.
+
+### Mutations
+
+If running with the read/admin configuration, initial assignments will be needed, for the case when users set e.g. when users assign: `READ`, `ADMIN`
+
+Some of the below are not currently available to users but including them here for consideration. This is also a fairly substantial list, which makes the case for read and admin pre-set access groups which would be easier for users and sites to configure, rather than defining a long list of mutations per user/group.
+
+As a springboard for discussion, defaults could be assigned as follows:
+
+### Current Mutations
+
+| Operation | READ | ADMIN | ALL |
+| :---     |:---: |:---: |---: |
+Broadcast| |x|x|
+Ext-trigger| |x|x|
+Hold| |x|x|
+Kill| |x|x|
+Message| |x|x|
+Pause| |x|x|
+Ping|x| |x|
+Play| |x|x|
+Poll| |x|x|
+Read|x| |x|
+Release| |x|x|
+ReleaseHoldPoint| |x|x|
+Reload| |x|x|
+Remove| |x|x|
+Resume| |x|x|
+SetGraphWindowExtent| |x|x|
+SetHoldPoint| |x|x|
+SetOutputs| |x|x|
+SetVerbosity| |x|x|
+Stop| |x|x|
+Trigger| |x|x|
+
+### Future Mutations
+
+| Operation | READ | ADMIN | ALL |
+| :---     |:---: |:---: |---: |
+Cat-log|x| |x|
+Check-versions|x| |x|
+Clean| |x|x|
+Compare| |x|x|
+Config|x| |x|
+Diff| |x|x|
+Dump| |x|x|
+Edit| |x|x|
+Get-cylc-version|x| |x|
+Get-workflow-version|x| |x|
+Graph|x| |x|
+Install| |x|x|
+List|x| |x|
+Reinstall| |x|x|
+Report-timings|x| |x|
+Scan|x| |x|
+Search|x| |x|
+Show|x| |x|
+Terminal Access| |x|x|
+Workflow-state|x| |x|
+Validate|x| |x|
+View|x| |x|
+
+In code, `READ`, `ADMIN` and `ALL` will expand to the assigned operations.
+
+Any mutation without a specified access assignment will be denied by default.
+As future features are added, they will also need to be categorised.
+
+The associated arguments for the mutations may need consideration.
+
+### Security Check for Permission Level on Config Files
+
+We should set a recommended write permissions level for the user config file. It could pose a security risk to have users config files writable by others. Perhaps we could have a warning in the log if the file permissions are not strict enough.
```

Comment:
We should probably log a warning that the file is not sufficiently secure (writable by others) and fall back to locked down permissions (i.e., owner only).

Reply:
Proposal updated.